colcon_cmake/task/cmake/__init__.py

```diff
@@ -262,23 +260,9 @@ def get_cmake_version():
 
 def _parse_cmake_version():
     """
-    Parse the CMake version number by running `cmake --version`.
-
-    The version number is parse from the output of 'cmake --version' and is
-    expected in the form 'cmake version #.#.#' where each '#' character
-    represents part of the version number.
-
-    Additional text may follow, however only the first line is parsed.
-
-    Parsing is performed by the pkg_resources package and the returned object
-    is a Version object from that package.
-
-    This function blocks on the execution of cmake and should only be used to
-    cache the CMake version number.
-
-    External API users should use get_cmake_version()
+    Parse the CMake version printed by `CMAKE_EXECUTABLE --version`.
```

Comment:
Imo some of the sentences where not helpful, e.g.:

> This function blocks on the execution of cmake and should only be used to cache the CMake version number.

Since the function is not `async` it clearly is blocking. I don't see why that should be part of the docblock.

---

> Parsing is performed by the pkg_resources package and the returned object is a Version object from that package.

The return type clearly documents how the version is represented.

---

> The version number is parse from the output of 'cmake --version' and is expected in the form 'cmake version #.#.#' where each '#' character represents part of the version number.
> Additional text may follow, however only the first line is parsed.

The part - that the output of `cmake --version` is parsed - is imo the important part for the reader. The other text isn't relevant to the user of the API - at most it could be a comment documenting how the logic works if a developer wants to read / understand the implementation of the function.

Reply:
Ok. Again, I'm not going to hold the PR up on these details.